src/ESLint.Formatter/sarif-with-rules.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * @fileoverview SARIF v2.0 formatter
+ * @author Chris Meyer
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Helper Functions
+//------------------------------------------------------------------------------
+
+/**
+ * Returns the severity of warning or error
+ * @param {Object} message message object to examine
+ * @returns {string} severity level
+ * @private
+ */
+function getResultLevel(message) {
+    if (message.fatal || message.severity === 2) {
+        return "error";
+    }
+    return "warning";
+}
+
+
+//------------------------------------------------------------------------------
+// Public Interface
+//------------------------------------------------------------------------------
+
+module.exports = function(results, rules) {
+    const sarifLog = {
+        version: "2.0.0",
+        $schema: "http://json.schemastore.org/sarif-2.0.0",
+        runs: [
+            {
+                tool: {
```

Comment:
>tool [](start = 16, length = 4)

do we not have access to current ESLint version information that we could populate?

Reply:
Have sent this question out already, waiting for answer

---
In reply to: [249138670](https://github.com/Microsoft/sarif-sdk/pull/1222#discussion_r249138670) [](ancestors = 249138670)

Reply:
This information isn't available.

---
In reply to: [249141099](https://github.com/Microsoft/sarif-sdk/pull/1222#discussion_r249141099) [](ancestors = 249141099,249138670)